creusot-contracts/src/lib.rs

```diff
@@ -156,9 +174,38 @@ pub mod std;
 #[cfg(creusot)]
 pub mod num_rational;
 
+#[cfg(creusot)]
+pub mod ghost;
+
 #[cfg(creusot)]
 pub mod snapshot;
 
+#[cfg(not(creusot))]
+pub mod ghost {
+    pub struct GhostBox<T>(std::marker::PhantomData<T>)
+    where
+        T: ?Sized;
+
+    impl<T: ?Sized> GhostBox<T> {
+        #[doc(hidden)]
+        pub fn from_fn(_: fn() -> T) -> Self {
+            GhostBox(std::marker::PhantomData)
+        }
+
+        pub fn uninit() -> Self {
+            Self(std::marker::PhantomData)
+        }
+    }
+
+    impl<T: ?Sized> Clone for GhostBox<T> {
+        fn clone(&self) -> Self {
+            GhostBox(std::marker::PhantomData)
+        }
+    }
+
+    impl<T: ?Sized> Copy for GhostBox<T> {}
+}
```

Comment:
I think we should move this outside of `lib.rs`, either into the existing `ghost.rs` or into a `ghost_no_creusot.rs` or something.